### PR TITLE
Create full screen view option in openQA

### PR DESCRIPTION
**Pull request by krauselukas**
Added a "fullscreen" checkbox to the filtersection of the index page. Currently it just hide's the "welcoming box" and the footer.

### CODE REVIEW

File: templates/layouts/bootstrap.html.ep [hunk 1]
```diff
@@ -132,6 +132,9 @@
           %= content
       </div>
 
+    % my $fullscreen = $self->param('fullscreen');
+
+    % if (!$fullscreen) {
```
Comment (coolo) [inline templates/layouts/bootstrap.html.ep]: nope. If you need a different layout for a specific template, specify that in the template

File: templates/main/index.html.ep [hunk 2]
```diff
@@ -7,6 +7,9 @@
     setupIndexPage();
 % end
 
+% my $fullscreen = $self->param('fullscreen');
+
+% if (!$fullscreen) {
```
Comment (coolo) [inline templates/main/index.html.ep]: split this into a partial then

Comment (krauselukas) [inline templates/main/index.html.ep]: @coolo allright, i will do that soon!

**Comment by krauselukas** [comment]
screenshot_2017-04-19_13-48-19
Here is the checkbox to enable the fullscreen view in openQA.

screenshot_2017-04-19_13-49-06
After enabling, the navbar at the top and the footer on the bottom will disappear to create more space.

screenshot_2017-04-19_13-50-02
If you move the mouse courser to the top the navbar will appear again.

File: templates/layouts/bootstrap.html.ep [hunk 3]
```diff
@@ -128,10 +129,14 @@
               </div><!-- /.navbar-collapse -->
           </div>
       </nav>
+    </div>
       <div class="container" id="content">
           %= content
       </div>
 
+    % my $fullscreen = $self->param('fullscreen');
```
Comment (Martchus) [inline templates/layouts/bootstrap.html.ep]: You need to stash this variable in the controller (Main.pm), so you don't have to define it here again.

File: assets/javascripts/fullscreen.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  document.addEventListener('mousemove', function(e){
+    mouseY = e.clientY || e.pageY;
+    if (mouseY < navbarHeight) {
+      $("#nav-main").removeClass('hidden').addClass('show');
+    }
+    else if (mouseY > navbarHeight) {
+      $("#nav-main").removeClass('show').addClass('hidden')
+    }
+  }, false);
+}};
```
Comment (Martchus) [inline assets/javascripts/fullscreen.js]: Please fix the indentation.

**Comment by krauselukas** [comment]
screenshot_2017-04-21_15-57-50
Decreased the margin to increase the size of the builds

File: assets/sass/fullscreen.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#content_fullscreen {
```
Comment (coolo) [inline assets/sass/fullscreen.scss]: please do not add a new file for 5 lines.

File: assets/javascripts/fullscreen.js (hunk 4 above)
Comment (coolo) [inline assets/javascripts/fullscreen.js]: same here - do not add a new file for one function

File: templates/layouts/bootstrap.html.ep [hunk 6]
```diff
@@ -40,6 +40,7 @@
 
   </head>
   <body>
+    <div id="nav-main">
```
Comment (coolo) [inline templates/layouts/bootstrap.html.ep]: why do you need a div around the nav ? and why do you need a id anyway? There aren't that many 'nav' elements around

File: templates/main/index.html.ep [hunk 7]
```diff
@@ -70,7 +69,11 @@
                 %>
 
                 <input value="1" name="fullscreen" type="checkbox" id="filter-fullscreen">
+<<<<<<< HEAD
```
Comment (coolo) [inline templates/main/index.html.ep]: you generally want to avoid commiting these characters

Comment (krauselukas) [inline templates/main/index.html.ep]: yes, sorry missed that part

**Comment by coolo** [comment]
please do not add WIP to every commit. That is rather pointless - most if not all commits done is work in a progress. Otherwise it's BS:

**Comment by krauselukas** [comment]
@coolo okay, thanks for letting me know! its not WIP anymore, so i had to remove it anyway :)

File: lib/OpenQA/WebAPI/Controller/Main.pm [hunk 8]
```diff
@@ -52,7 +52,8 @@ sub index {
     my $only_tagged = $self->param('only_tagged') // 0;
     my $show_tags   = $self->param('show_tags')   // $only_tagged;
     my $group_params = $self->every_param('group');
-
+    my $fullscreen   = $self->param('fullscreen');
+    $self->stash('fullscreen', $fullscreen);
```
Comment (mudler) [inline lib/OpenQA/WebAPI/Controller/Main.pm]: not sure this is needed, [you can access to params into templates as well](http://mojolicious.org/perldoc/Mojolicious/Plugin/DefaultHelpers#param)

File: templates/main/index.html.ep [hunk 9]
```diff
@@ -19,7 +20,7 @@
     </div>
 </div>
 
-<div id="build-results"></div>
+<div id="build-results">
```
Comment (coolo) [inline templates/main/index.html.ep]: not sure how this is related

File: lib/OpenQA/WebAPI/Controller/Main.pm (hunk 8 above)
Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/Main.pm]: I can't see the stash used either

Comment (krauselukas) [inline lib/OpenQA/WebAPI/Controller/Main.pm]: yeah it is not needed anymore, i used a separate template for full screen when i started this.

File: templates/main/index.html.ep (hunk 9 above)
Comment (krauselukas) [inline templates/main/index.html.ep]: I will change this

**Pull request merged**